Add release notes for GOV.UK Prototype Kit v9.13.0 (#1013)
alphagov/govuk-prototype-kit: CHANGELOG.md

# Unreleased

# 9.13.0 (Feature release)

> ## New features

Reviewer: We'll also need to include 'Update to GOV.UK Frontend v3.12' in the new features section – the PR for that will come later, just before we do the release.

Normally we just include it as a bullet item ([like in 9.12](https://github.com/alphagov/govuk-prototype-kit/blob/master/CHANGELOG.md#9120-feature-release)) but I'm not sure how that would work when we have an h3 with copy like we do here.

### Opt in to the new GOV.UK Frontend link styles

Links now have underlines that are consistently thinner and a bit further away from the link text.

Links also have a clearer hover state, where the underline gets thicker to make the link stand out to users.

> The new link styles are opt-out by default because [Chromium browsers currently cannot display them](https://github.com/alphagov/govuk-frontend/issues/2204).

Reviewer: Great idea linking to the known issue, but I don't think this summary is quite right – Chromium browsers can display them, it's just there's an issue with links that are inside a multi-column layout.

I'm also tripping up over 'opt-out by default' – I'd personally would have said they're opt-in by default! I think there's ambiguity as to whether this means 'by default **you are** opted out' or 'by default **you have to** opt-out'?

Reviewer: I think this comment still needs addressing.

Author: Ah, sorry - I tried to address it in [this commit](https://github.com/alphagov/govuk-prototype-kit/pull/1013/commits/88b79f46a2c2b5c173e1f12ab807762d79eb4044) a few hours ago.

[Read more about the new link styles in the GOV.UK Frontend release notes.](https://github.com/alphagov/govuk-frontend/releases)

This was added in [Pull request #1012: Implement the new link and hover styles in the Prototype Kit](https://github.com/alphagov/govuk-prototype-kit/issues/1012).

## Fixes

> - [Pull request #990: Make Sass errors clearer to the user](https://github.com/alphagov/govuk-prototype-kit/pull/990)

Reviewer: I know @joelanman added this as a fix in #990, but I'm wondering if this is worth listing as a new feature with a little bit more information. There's a change to the way the kit works when there's an error in a user's Sass – which could potentially be a little confusing for users who are used to the old way it worked?

# Unreleased

# 9.13.0 (Feature release)

## New features

### Make Sass errors clearer to users

> Previously, the Sass app continued to run even after Sass stopped compiling because of an error.

Reviewer: This section is great, but there's a couple of things that I think could be clearer:

1. I'm not sure a user would know what we mean by 'Sass app'
2. Does saying 'previously, the Sass app continued to run' suggest that the new behaviour is that it will stop running?

This might be a little long, but what about something like:

```suggestion
Previously, it wasn't obvious to users if Sass has stopped updating because of an error. An error would be printed to the Terminal, but nothing would happen in the browser. We know that it was easy to miss, and could be confusing.
```

Author: @36degrees Your suggestion looks good! Just a couple of questions about:

> We know that it was easy to miss, and could be confusing.

- By "it," do we mean the error notification or the error itself?
- What do we mean by "confusing" - the error / error notification or the user experience of Sass not updating without their knowing why?

Could a slight rewording like the below work?

> Previously, it was not obvious to users if Sass has stopped updating because of an error. An error would be printed to the Terminal, but nothing would happen in the browser. We know that this error was easy to miss, and could cause confusion.

> Previously, it was not obvious to users if Sass has stopped updating because of an error. An error would be printed to the Terminal, but nothing would happen in the browser. We know that this error was easy to miss, which could cause confusion.

Author: ```suggestion
Previously, it was not obvious to users if Sass had stopped updating because of an error. An error would be printed to the Terminal, but nothing would happen in the browser. We know that this error was easy to miss, which could cause confusion.
```

Now, when there's a Sass error, it creates a blank `application.css` file to make the site look broken to users.

If you fix the Sass error, the site will automatically reload.

We have also changed the Gulp log level to be quieter, so that errors stand out.

> This was added in [Pull request #990: Make Sass errors clearer to the user] (https://github.com/alphagov/govuk-prototype-kit/pull/990).

Reviewer: ```suggestion
This was added in [Pull request #990: Make Sass errors clearer to the user](https://github.com/alphagov/govuk-prototype-kit/pull/990).
```

> We have also changed the [Gulp](https://www.npmjs.com/package/gulp-sass) log level to be quieter, so that errors stand out.

Author: ```suggestion
We have also changed the [Gulp](https://www.npmjs.com/package/gulp-sass) log level to be less detailed, so that errors stand out.
```

### Opt in to the new GOV.UK Frontend link styles

Links now have underlines that are consistently thinner and a bit further away from the link text.

Links also have a clearer hover state, where the underline gets thicker to make the link stand out to users.

> The new link styles are opt-out by default because [Chromium browsers currently cannot display them](https://github.com/alphagov/govuk-frontend/issues/2204).

Author: ```suggestion
The new link styles are opt-in by default because [Chromium browsers currently cannot display them](https://github.com/alphagov/govuk-frontend/issues/2204).
```

> The new link styles are opt-in by default because [Chromium browsers currently cannot display them](https://github.com/alphagov/govuk-frontend/issues/2204).

Author: ```suggestion
The new link styles are opt-in because [Chromium browsers have an issue with displaying them in a multi-column layout](https://github.com/alphagov/govuk-frontend/issues/2204).
```

> The new link styles are opt-in because [Chromium browsers have an issue with displaying them in a multi-column layout](https://github.com/alphagov/govuk-frontend/issues/2204).

Author: ```suggestion
The new link styles are opt-in because [Chromium browsers have an issue with displaying them inside a multi-column layout](https://github.com/alphagov/govuk-frontend/issues/2204).
```

> The new link styles are opt-in because [Chromium browsers have an issue with displaying them inside a multi-column layout](https://github.com/alphagov/govuk-frontend/issues/2204).

Author: ```suggestion
The new link styles are opt-in because [Chromium browsers have an issue with links inside a multi-column layout](https://github.com/alphagov/govuk-frontend/issues/2204).
```

> [Read more about the new link styles in the GOV.UK Frontend release notes.](https://github.com/alphagov/govuk-frontend/releases)

Author: ```suggestion
Read more about the new link styles in the [GOV.UK Frontend release notes](https://github.com/alphagov/govuk-frontend/releases).
```

This was added in [Pull request #1012: Implement the new link and hover styles in the Prototype Kit](https://github.com/alphagov/govuk-prototype-kit/issues/1012).

### Update to GOV.UK Frontend

> This was added in Pull request #???: Update to GOV.UK Frontend v3.12.

Author: Need to add link to pull request.
